crates/ruff/src/rules/pyupgrade/rules/unnecessary_coding_comment.rs

```diff
@@ -55,20 +55,40 @@ pub(crate) fn unnecessary_coding_comment(
 ) {
     // The coding comment must be on one of the first two lines. Since each comment spans at least
     // one line, we only need to check the first two comments at most.
-    for range in indexer.comment_ranges().iter().take(2) {
-        let line = locator.slice(*range);
-        if CODING_COMMENT_REGEX.is_match(line) {
+    for comment_range in indexer.comment_ranges().iter().take(2) {
+        let line_range = locator.full_line_range(comment_range.start());
+        let line = locator.slice(line_range);
+        let comment = locator.slice(*comment_range);
+
+        // Both the line and the comment itself must match to prevent false positives
+        // where the comment is on a line that includes content that looks like a coding comment
+        // or the coding comment is preceded by non-whitespace
+        if CODING_COMMENT_REGEX.is_match(line) && CODING_COMMENT_REGEX.is_match(comment) {
```

Comment:
I think I would still recommend doing something like:

```rust
// Check if there's any content preceding the comment.
let line_start = locator.line_start(*range.start());
if !locator
    .slice(TextRange::new(line_start, range.start()))
    .trim()
    .is_empty()
{
    continue;
}
```

Personally, I find that much easier to understand than running the regex twice.